### ISSUE TITLE
Regarding CHANGELOG

### ISSUE THREAD

**Issue by D3DeFi** [ansible-collections/community.zabbix#52]
##### SUMMARY
Its apparent that at some point we will be releasing next version of this collection. In the meantime, changes are accumulating. 

Should we require that with each change to a module or roles, an entry will be added to the CHANGELOG.md under "`devel`" section or something like that?

##### ISSUE TYPE
- Documentation Report

**Comment by sky-joker** [comment]
I wonder how they manage the other collections.
I took a quick look at it, but it's stored in the traditional `changelogs/fragments`, or it's using a  Changelog file.

**Comment by dj-wasabi** [comment]
I manually updated the changelog md when creating a new release. Was fine for the amount of PR's that I had with the roles, but I was just not motivated enough to create a (python) script for it to automate it. I don't know how it will be when also the modules are part of the repo?

**Comment by D3DeFi** [comment]
> 
> 
> I don't know how it will be when also the modules are part of the repo?

I cannot say if development of the modules will speed up in the future, but when the modules were residing in ansible/ansible, it was very slow process. Less than 5 patches in a month, sometimes no patches at all. I, for example, would like to be more active in the future, but I am not sure if I will have more time for this collection.

Maybe if new maintainers join our ranks in the future, development of both roles and modules will demand something automatic, but for now it seems that it will be easier to add changelog entries before publish.

We can for example have version `devel` in the top of a _CHANGELOG.md_ or something like that and add entries to it with each feature/bugfix PR. Once it is ready to be tagged as a new release, we simply rename the `devel` to a specific version.

**Comment by sky-joker** [comment]
> So if you agree, lets stay with this manual approach and see how it turns out in the future. 

I agree :)  
First of all,  let's make it a policy to update CHANGELOG manually.

**Comment by D3DeFi** [comment]
I will add this to the (prepared) contribution rules

**Comment by sky-joker** [comment]
FYI
I found this tool.

https://github.com/ansible-community/antsibull-changelog

It's unclear if Collections uses this tool as a standard.

**Comment by D3DeFi** [comment]
@gundalow will we be *advised* to use this in the future?

**Comment by gundalow** [comment]
If you add fragments then the above script will do the rest for you. We are using this in community.general, community.network and someothet repos.
No requirement for you to use it, though I'd hope it would make your life easier, ie no need to to have changelog.rst being on continual merge conflict.

**Comment by D3DeFi** [comment]
> If you add fragments then the above script will do the rest for you. We are using this in community.general, community.network and someothet repos.
> No requirement for you to use it, though I'd hope it would make your life easier, ie no need to to have changelog.rst being on continual merge conflict.

Sounds good. Reopening this

**Comment by gundalow** [comment]
Just to give a bit more detail around this. My definition collections are separate and free to do things differently to gh/ansible/ansible.

That said, I don't expect every single collection owner to have to setup their own GitHub workflow, CI, documentation, changelog, etc, etc. So we (the Ansible Community team + wider Ansible community) are building and documenting one way of doing things that we'd hope would work for most people. 

However, what we do for community.general + community.network, which are huge collections of 800+400 modules respectively aren't necessarily right for a collection like Zabbix.


If you have any questions or feedback about https://github.com/ansible-community/antsibull-changelog please do raise issues in that repo.

**Comment by sky-joker** [comment]
Thank you @gundalow :)  
I'll take the information you've given me as a guide.

**Comment by D3DeFi** [comment]
So it seems that this is changed and ansible now requires changelog to be in a specific format. I am starting on reworking this to make antsibull-changelog work for us too.

**Comment by D3DeFi** [comment]
So the thing with antsibull-changelog is that it doesn't care about roles, neither does ansible in its Ansible Community Distribution changelogs. I am not really interested in keeping separate files for the two and want it unified, but at the same time it would be unwise to force changes regarding roles to the upstream ansible itself.

What do we do here? Give me ideas

**Comment by gundalow** [comment]
antsibull-changelog should just work for you, ie same changelog process for plugins and roles.

**Comment by felixfontein** [comment]
@D3DeFi what do you mean with "doesn't care about roles"? You can mention changes in roles in regular changelog entries, can't you? Or do you mean announcing new roles? This is not supported right now because there is no way to identify new roles (AFAIK, I'm not very familiar with roles in collections yet). (The same is true for filter and test plugins, since they are not documentable yet.)

**Comment by felixfontein** [comment]
(I've created ansible-community/antsibull-changelog#37 to track support for explicitly reporting new roles, tests, filters, etc.)

**Comment by D3DeFi** [comment]
> @D3DeFi what do you mean with "doesn't care about roles"? You can mention changes in roles in regular changelog entries, can't you? Or do you mean announcing new roles? This is not supported right now because there is no way to identify new roles (AFAIK, I'm not very familiar with roles in collections yet). (The same is true for filter and test plugins, since they are not documentable yet.)

@felixfontein I am sorry, maybe I should have used more proper wording. What I meant was that during familiarizing myself with antsibull-changelog I've noticed that only plugins are being mentioned.

This lead to a confusion on my side if you add to it the fact that in 2.9.x releases there were only core functionality and plugins mentioned in the changelogs. And now with 2.10.0 there would suddenly appear role entries.

So, is this correct? Lets say I will put 5x (we have 5 roles rn) minor_changes entry into 0.2.0 release 'role zabbix_xyz from dj-wasabi/ansible-zabbix-xyz was migrated to the community.zabbix collection' and then, there will be bugfixes and minor_changes entries present for roles in the 0.3.0 and so on.

**Comment by felixfontein** [comment]
If you add changelog fragments to this repo which mention roles, these will get included. antsibull-changelog does not care about the contents of the changelog fragments, it trusts the user to put meaningful content in there. So if you put `role zabbix_xyz from dj-wasabi/ansible-zabbix-xyz was migrated to the community.zabbix collection` (not sure though how that's related to this collections changelog?) into a fragment for this collection, that text will show up under `Minor Changes` in the generated changelog.

**Comment by D3DeFi** [comment]
Thank you for clarification. Regarding "role zabbix_xyz from dj-wasabi/ansible-zabbix-xyz was migrated to the community.zabbix collection" - this may not be the best formulation, but I feel we should mention this information somewhere in the changelog as roles were not part of this collection until 0.2.0.

Also as you have said antsibull-changelog is currently not able to detect new roles and I feel obliged to mention them.

So maybe something under minor_changes as "- new role zabbix_agent - previously dj-wasabi/zabbix-agent"? What do you think about that? I should be able to provide PR with new changelog later today

**Comment by felixfontein** [comment]
Ah, sorry, I completely misunderstood `role zabbix_xyz from dj-wasabi/ansible-zabbix-xyz was migrated to the community.zabbix collection` - I thought that the role in another repository was adjusted to use this collection (instead of ansible <= 2.9). But I now think you mean that the role was moved from that repository into this collection. That's definitely related to this collection's changelog.

I'd add it as `minor_changes`, the way you suggested (`new role zabbix_agent - previously dj-wasabi/zabbix-agent`). That should be good.

**Comment by D3DeFi** [comment]
Can you guys please check if #179 looks good for you?

**Comment by felixfontein** [comment]
@D3DeFi antisbull-changelog works with RST, so you can use RST links (```... `this is a link <https://example.com>`_ ...```)
